web/services/createProjectsService.php

```diff
@@ -208,11 +209,34 @@
 
 
     if (count($createProjects) >= 1)
-        if (ProjectsFacade::CreateProjects($createProjects) == -1)
-            $string = "<return service='createProjects'><error id='1'>There was some error while creating the projects</error></return>";
+        if(count($createProjects) == 1){
+            $operationResult = ProjectsFacade::CreateProject($createProjects[0]);
+            if (!$operationResult->getIsSuccessful()) {
+                http_response_code($operationResult->getResponseCode());
+                $string = "<return service='createProjects'><error id='" . $operationResult->getErrorNumber() . "'>" . $operationResult->getMessage() . "</error></return>";
+            }
+        }
+        else {
+            //leaving the possibility to create multiple projects in the future
+            $operationResults = ProjectsFacade::CreateProjects($createProjects);
+            $errors = array_filter($operationResults, function ($item) {
+                return (!$item->getIsSuccessful());
+            });
+            if($errors){
+                //if multiple failures, let's just return a 500
+                http_response_code(500);
+                $string .= "<return service='createProjects'><errors>";
```

Comment:
This produces an "undefined variable" error, because it's the first time you use `$string` in this code path. Just use the `=` operator instead of `.=` :slightly_smiling_face:

Reply:
Ah! Good catch. Will definitely fix!